File: project-euler/ruby/10.rb
#!/usr/bin/env ruby

def prime(n)
	(2..n**0.5).each do |a|
		return false if n % a == 0
	end
	return true
end

sum = 2
n   = 3

# while is ~10% faster, but (...).step is perhaps more idiomatic.
#(3..2000000).step(2) do |n|
while n < 2000000
	sum += n if prime(n)
	n += 2
end

puts sum
# 142913828922

File: project-euler/python/10.py
#!/usr/bin/env python

# Sum of primes below 2,000,000

from lib import divisible_by_any

# Unfortunately slow, but a generator-sieve overflows the recursion limit.
def sum_primes(limit):
	n = 3
	primes = [2]

	while n < limit:
		if not divisible_by_any(n, primes):
			primes.append(n)
		n += 2
	return sum(primes)

if __name__ == '__main__':
	limit = 2000000
	print "Answer: " + str(sum_primes(limit))

